Gravity/Gravity.NUnit.Integration/RSAPI_IntegrationTest.cs

```diff
@@ -0,0 +1,395 @@
+                    case RdoFieldType.YesNo:
+                        newObjectValue = field.ValueAsYesNo;
+                        break;
+                    case RdoFieldType.Currency:
+                        newObjectValue = field.ValueAsCurrency;
+                        break;
+                    case RdoFieldType.Decimal:
+                        newObjectValue = field.ValueAsDecimal;
+                        break;
+
+                }
+
+                Console.WriteLine("Act Complete....");
+
+                //Assert
+                Console.WriteLine("Starting Assertion....");
+
+                //Assert
+                Assert.AreEqual(sampleData, newObjectValue);
+
+                Console.WriteLine("Assertion Complete....");
+            }
+            catch (Exception ex)
+            {
+                throw new Exception("Error encountered in " + System.Reflection.MethodBase.GetCurrentMethod().Name, ex);
+            }
+            finally
+            {
+                Console.WriteLine("Ending Test case " + System.Reflection.MethodBase.GetCurrentMethod().Name);
+            }
+        }
+
+        [TestCase("LongTextField", TestValues.LongTextFieldValue)]
+        [TestCase("FixedTextField", TestValues.String100Length)]
+        [TestCase("IntegerField", -1)]
+        [TestCase("BoolField", true)]
+        [TestCase("DecimalField", 123.45)]
+        [TestCase("CurrencyField", 5648.54)]
+        public void Valid_Gravity_Object_Read_Field_Type<T>(string objectPropertyName, T sampleData)
+        {
+            Console.WriteLine(System.Reflection.MethodBase.GetCurrentMethod().Name + " Created");
+
+            try
+            {
+                //Arrange
+                Console.WriteLine("Starting Arrangement for property...." + objectPropertyName);
+
+                GravityLevelOne testObject = new GravityLevelOne();
+                testObject.Name = "TestObjectRead_" + objectPropertyName + Guid.NewGuid().ToString();
+
+                Guid testObjectTypeGuid = testObject.GetType().GetCustomAttribute<RelativityObjectAttribute>(false).ObjectTypeGuid;
+                Guid nameFieldGuid = Gravity.Base.BaseDto.GetRelativityFieldGuidOfProperty<GravityLevelOne>("Name");
+                Guid testFieldGuid = Gravity.Base.BaseDto.GetRelativityFieldGuidOfProperty<GravityLevelOne>(objectPropertyName);
+                RdoFieldType fieldType = Gravity.Base.BaseDto.GetRelativityFieldTypeOfProperty<GravityLevelOne>(objectPropertyName);
+
+                _client.APIOptions.WorkspaceID = _workspaceId;
+
+                var dto = new RDO();
+                int newArtifactId = -1;
+                dto.ArtifactTypeGuids.Add(testObjectTypeGuid);
+                dto.Fields.Add(new FieldValue(nameFieldGuid, testObject.Name));
+                dto.Fields.Add(new FieldValue(testFieldGuid, sampleData));
+                WriteResultSet<RDO> writeResults = _client.Repositories.RDO.Create(dto);
+
+                if (writeResults.Success)
```

Comment:
We already have a helper method for processing WriteResults (including logging all errors, if I recall correctly). That should be used instead.

Reply:
agree with the logging, but think this has to do with logging to the automated testing tools.  Will verify with SETs in the AM.